src/register-evm-agent/src/register_evm_agent/src/reservation.rs

```diff
@@ -62,8 +64,26 @@ impl<'a> ReservationService<'a> {
             self.mint_native_tokens_to_address(amount_to_mint).await?;
         }
 
+        let nonce = self.client.account_basic(address.clone()).await?.nonce;
+
+        let tx = TransactionBuilder {
+            from: &address.clone(),
+            to: Some(address),
+            nonce,
+            value: U256::zero(),
+            gas: 23_000_u64.into(),
+            gas_price: None,
+            input: self.reserve_canister_id.as_slice().to_vec(),
+            signature: SigningMethod::SigningKey(self.wallet.signer()),
+            chain_id: DEFAULT_CHAIN_ID,
+        }
+        .calculate_hash_and_build()?;
+
+        info!("sending transaction to reserve address...");
+        let tx_hash = self.client.send_raw_transaction(tx).await??;
+
         self.client
-            .reserve_address(self.reserve_canister_id, address)
+            .reserve_address(self.reserve_canister_id, tx_hash)
```

Comment:
before reserving you should wait for the TX to be processed. For example, by getting the status in a loop every second

Reply:
ah, Yes